feat(deps): add devnet-slots-in-an-epoch argument

fmoura opened the pull request.

gligneul committed:
refactor: remove unnecessary Rust code

Only the authority-claimer service will remain in the Node 2.0.

gligneul committed:
refactor: simplify CI workflows

Move the jobs from the assess-code-quality workflows to the build
workflow.

gligneul committed:
chore: remove server-manager

gligneul committed:
feat: bump contracts to 2.0-rc.2

marcelstanley committed:
fix(test): remove config for host mode

marcelstanley committed:
chore(test): run end-to-end tests via make

fmoura committed:
chore: Bump Rust to 1.78.0

Added general '-A clippy::mixed_attributes_style' to CI clippy execution to avoid linter warning on the automatically generated file 'iconsensus.rs'

GMKrieger committed:
feat(graphql): add postraphile to supervisor

torives committed:
feat: add merkle package

Adds the merkle.CreateProofs function that will be used to generate the cryptographic proofs for the rollup outputs

gligneul committed:
fix(claimer): parse iconsensus address

fmoura committed:
feat: Add CI Custom Image

marcelstanley committed:
chore: fix typos

fmoura committed:
feat: Add --no-mining opt to Devnet

renan061 committed:
refactor: extract linewriter to package

renan061 committed:
feat: add tooling for creating snapshots programmatically

renan061 committed:
feat: add rollupsmachine package

renan061 committed:
fix: stop end-to-end testing

renan061 committed:
chore: update machine version and devnet addresses

renan061 committed:
feat: update the Dockerfile with emulator dependencies

- Adds linux.bin and rootfs.ext2 download to the emulator stage.
- Adds a emulator-devel stage that install libcmt and xgenext2fs.
- Configures the CI image to use CGO.
- Refactors the Dockerfile to improve readability.

Commit:
feat: add machine emulator C API bindings

renan061 committed:
test: add tests to the emulator bindings

marcelstanley committed:
feat: normalize bool config parameters

Applied suffixes ENABLED to boolean parameters to match the node
internal state and set their appropriate default values.

marcelstanley committed:
fix: adjust boolean env vars after normalization

GMKrieger committed:
fix: update postgraphile command and lock version

GMKrieger committed:
feat(claimer): add claimer repository

GMKrieger committed:
feat(claimer): remove broker and redis

marcelstanley committed:
fix(build): update boolean env vars usage

GMKrieger committed:
feat(data): add base repository migrations

GMKrieger committed:
feat(node): update node models

GMKrieger committed:
feat(data): add base repository functions

GMKrieger committed:
feat(data): add base repository tests

GMKrieger committed:
feat(data): add validator repository functions

GMKrieger committed:
feat(data): add validator repository tests

GMKrieger committed:
feat(data): add input reader repository functions

GMKrieger committed:
feat(data): add input reader repository tests

GMKrieger committed:
feat(node): add run migrations to node startup

GMKrieger committed:
feat(data): add postgraphile views

marcelstanley committed:
chore: fix file format

fmoura committed:
fix(tests): update port usage for testcontainers

fmoura committed:
ci: disable reaper to avoid failure on testcontainers

This issue seems related to
https://github.com/testcontainers/testcontainers-go/issues/2172

fmoura committed:
fix(tests): mine blocks while adding inputs

GMKrieger committed:
feat(data): add repository checks for numerics

fmoura committed:
feat(cli): Add '--verbose ' option to all commands

fmoura committed:
chore(ethutil): Remove unused arguments

fmoura committed:
feat(util): Add retry policy utility function

fmoura committed:
feat(cli): Add disable postgres or devnet to run-deps

fmoura committed:
feat(cli): Add add and list applications commands

fmoura committed:
feat(cli): Add check-version and upgrade schema commands

fmoura committed:
feat(evm-reader): Add evm-reader

fmoura committed:
feat(evm-reader): Add evm-reader tests

fmoura committed:
feat(evm-reader): Add evm-reader command

fmoura committed:
chore: fix Dockerfile non matching case

torives committed:
feat: add validator package

torives committed:
feat(validator): add service to the supervisor

torives committed:
feat(validator): run as a standalone binary

fmoura committed:
chore(devnet): Bump anvil version to nightly-626221f5ef44b4af950a08e09bd714650d9eb77d

fmoura committed:
feat(devnet): add devnet devnet-finalization-offset config

GMKrieger committed:
feat(node): add PostgresSslMode config

GMKrieger committed:
chore: remove obsolete config from yaml files

fmoura committed:
feat(db): Add Epoch and Snapshot

fmoura committed:
feat(db): Remove Claim

fmoura committed:
feat(db): Move epochLength from Application to NodeConfig

marcelstanley commented on cmd/cartesi-rollups-cli/root/deps/deps.go:
```diff
@@ -56,6 +56,10 @@ func init() {
 
+	Cmd.Flags().StringVar(&depsConfig.Devnet.SlotsInAnEpoch, "devnet-slots-in-an-epoch",
+		deps.DefaultSlotsInAnEpoch,
+		"Devnet slots in an epoch")
```
What's the exact purpose of adding this new argument?

Also, perhaps adding a longer explanation would make it easier to understand that we're referring to. See explanation [available in anvil](https://book.getfoundry.sh/reference/anvil/):
> To speed up the finalization of blocks, you can use the --slots-in-an-epoch flag with a value of 1 for example. This will lead to the block at height N-2 being finalized, where N is the latest block.

BTW, how does this work with `CARTESI_BLOCKCHAIN_FINALITY_OFFSET`?

fmoura replied:
I will improve the docs. 

The purpose is to , when testing Node against Devnet, and setting the Default Block to "finalized" you do not have to wait for the Anvil's default  64 blocks long epoch ( which will take about a minute to finalize)

fmoura replied:
About the `CARTESI_BLOCKCHAIN_FINALITY_OFFSET`. It is a "legacy" parameter, still in use by "Claimer". It has the same meaning as Anvil's `slots-in-an-epoch`, but not sure what are it's effects on Claimer behavior ( probably how many blocks it waits to consider claim submission transaction as finalized)

fmoura committed:
feat(repository): update validator methods

fmoura committed:
chore(evm-reader): Make code more go-like

renan061 committed:
refactor: add unit tests to the rollupsmachine

fmoura committed:
feat(evm-reader): Handle epochs

fmoura committed:
chore(build): fix docker-bake.hcl file format

fmoura committed:
feat(contracts): bump contracts version to 2.0.0-rc6

fmoura committed:
feat(config): Remove IConsensus and EpochLength from PersistentNodeConfig

renan061 committed:
feat: change emulator version

torives committed:
refactor(repository): update test name

GMKrieger committed:
feat(db): run upgrade via cli image

torives committed:
fix(startup): properly wrap error

fmoura committed:
chore(deps): Improve deps startup logs

fmoura committed:
chore(db): Make "no rows" log Debug

fmoura committed:
fix(util): Fix Retry Policy behavior

renan061 committed:
feat(advancer): add nodemachine

renan061 committed:
feat(advancer): add advancer service

renan061 committed:
refactor(advancer): add contexts and timeouts to rollupsmachine and nodemachine

torives committed:
test(validator): add repository integration tests

fmoura committed:
fix(evm-reader): Fix hidden error log

GMKrieger committed:
fix(build): start node after cli completed

fmoura committed:
fix(cli): Fix 'app add' command missing required flag

fmoura committed:
fix(config): Disable Postgres ssl mode at localhost

fmoura committed:
feat(evm-reader): store only non empty epochs

torives committed:
fix(repository): correct field order

When reading data from the returned database row in the GetPreviousEpoch function, `claimHash` was being assigned to `transactionHash` and vice-versa.

fmoura committed:
chore(evm-reader): Update log for clarity

renan061 committed:
refactor(repository): move migrations to folder and refactor repository

fmoura committed:
chore(deps): Remove Devnet 'BlockToWaitForOnStartup' as it is no longer needed

fmoura committed:
chore(evm-reader): adjust 'mostRecentBlock' mentions

torives committed:
fix(repository): get the previous epoch by index

Since epochs with no inputs won't be stored in the database, the block range between epochs might be disjointed, breaking this function. Filtering results based on the epoch index alone will be enough.

fmoura committed:
fix: fix the CI make-generate to use an image with the cartesi machine

torives committed:
fix(repository): add hash when inserting outputs

torives committed:
refactor(validator): correct error data

renan061 committed:
feat: move snapshot to the tooling folder and add db package

torives committed:
feat(validator): add nil machine claim error

All epochs have at least one input. If an input has no outputs, then Input.OutputsHash equals the last input OutputsHash. If it is the first input and it has no outputs, its OutputsHash is the application's template hash. Therefore, if this value is nil the node is in an inconsistent state.

fmoura committed:
fix(db): Fix Schema postgres ssl mode handling

vfusco commented:
Deprecated
